### PR TITLE
Revamp sight light text

### LINKED ISSUE #4549: [Refactoring]: Sight/Light descriptions difficult to translate
### Describe the problem

Refer to [Discord](https://discord.com/channels/296230822262865920/634664267190108171/1183100336844582982) and [thread](https://discord.com/channels/296230822262865920/634664267190108171/1183243155873726525)

Short version: The texts are long, unwieldy, difficult to translate, and break some good practises.
Paraphrasing - "Translation texts should be for short phrases only. Words and single sentences. No HTML formatting"

### The improvement you'd like to see

Refactor the description code away from a monolithic block and into simple, easy to translate chunks, with literal text marked to remain unchanged, .e.g. "width=y"

### Expected Benefits

Happier translation community relations.
Easier transition to improved dialog form where they can be used for tooltips.

### Additional Context

_No response_

### PR DESCRIPTION
**Pull request by bubblobill**
### Identify the Bug or Feature request
resolves #4549

### Description of the Change
Rewrote sight and light help texts to remove formatting and simplify translation components.
HTML is now constructed by substituting i18n text into HTML template string.

### Possible Drawbacks
A lot more individual key/values need to be translated.
Languages with no active translators will get the old text.

### Documentation Notes
Reformatted sight & light help texts for easier translation and easier transition to new UI.

### Release Notes
Reformatted sight & light help texts

- - -
This change is [Reviewable](https://reviewable.io/reviews/RPTools/maptool/4565)

### CODE REVIEW

File: src/main/java/net/rptools/maptool/client/ui/campaignproperties/CampaignPropertiesDialog.java [hunk 1]
```diff
@@ -529,4 +553,354 @@ public static void main(String[] args) {
...
+      parameters.putIfAbsent(
+          uncapitalize(capitalize(key, '.').replace(".", "")), I18N.getText(key));
+    }
+
+    /* Parameterised strings - need to be done individually */
+    parameters.put(
+        "wikiLinkReferral",
+        I18N.getText(
+            "sightLight.wikiLinkReferral",
+            "<i>wiki.rptools.info/index.php/Introduction_to_Lights_and_Sights</i>"));
```
Comment (irisiflimsi) [inline src/main/java/net/rptools/maptool/client/ui/campaignproperties/CampaignPropertiesDialog.java:590]: This format < i > sticks out like a sore thumb. Is it really necessary here?

File: src/main/java/net/rptools/maptool/client/ui/campaignproperties/CampaignPropertiesDialog.java [hunk 2]
```diff
@@ -529,4 +553,354 @@ public static void main(String[] args) {
...
+            <li>${structureListItemMultiple}<sup>1</sup></li>
+            <li>${structureListItemGroupName}</li>
+            <li>${structureListItemGroupedNames}</li>
+            <li>${structureListItemGroups}</li>
+            <li>${structureListItemSorting}</li>
+            """;
+    String syntaxHeading =
+        """
+              </ul>
+              <u><font size=5>${subheadingDefinitionSyntax}</font></u><br><br>
```
Comment (irisiflimsi) [inline src/main/java/net/rptools/maptool/client/ui/campaignproperties/CampaignPropertiesDialog.java]: I generally dislike the idea of having the closing syntax (here < / u l > in 658) in a separate string from the beginning (here < u l > in 641). This is side-effect programming and hard to follow and latter to maintain valid html, when names and other items change. Line 640 is better.

File: src/main/java/net/rptools/maptool/client/ui/campaignproperties/CampaignPropertiesDialog.java [hunk 3]
```diff
@@ -529,4 +553,354 @@ public static void main(String[] args) {
...
+            </tr>
+            <tr>
+              <th></th>
+              <th>${optionLabelLumens}</th>
+              <td><i>[${wordOptional}]</i>&nbsp;${optionDescriptionPersonalSightComponentLumens}<sup>4</sup></td>
+              <td${alignCellCenter}>+100</td>
+              <td${alignCellCenter}>+100</td>
+            </tr>
+            </table>
+            """;
```
Comment (irisiflimsi) [inline src/main/java/net/rptools/maptool/client/ui/campaignproperties/CampaignPropertiesDialog.java:817]: Same thing as above. If such blocks are necessary, have the opening and closing tags in separate strings (called blaOpen and blaClose). That removes the implicit order for the first and last text blocks.

File: src/main/java/net/rptools/maptool/client/ui/campaignproperties/CampaignPropertiesDialog.java [hunk 4]
```diff
@@ -529,4 +553,354 @@ public static void main(String[] args) {
...
+    }
+    return parameters;
+  }
+
+  /**
+   * Creates HTML for both sight and light help
+   *
+   * @return String[]
+   */
+  private String[] generateHelpText() {
```
Comment (irisiflimsi) [inline src/main/java/net/rptools/maptool/client/ui/campaignproperties/CampaignPropertiesDialog.java:633]: Maybe extract this into a helper class? The method length would suggest this and linters probably already complain about it.

File: src/main/resources/net/rptools/maptool/language/i18n.properties [hunk 5]
```diff
@@ -838,6 +838,127 @@ position.side.right         = Right
...
+light.example.text.aura.donutCone   = The same trick done with a cone.
+light.example.text.aura.donutHole   = This one hides a secret. If you have a range with no colour it becomes the range the light source begins. In this case the light begins at 20 and ends at 40.
+light.example.text.aura.gmRed       = A circular GM-only aura in red with a 15-foot diameter.
+light.example.text.aura.gmRedSquare = A 5-foot square red aura only visible to the GM.
+light.example.text.aura.lineOfSight = A beam 150 ft long with a width of 0.4 map units.
+light.example.text.aura.owner       = A circular green aura that is only visible to the token's owners.
+light.example.text.aura.rangeArcs   = A series of cones starting every 30 ft and ending 0.4 ft later.
+light.example.text.aura.rangeCircles= A series of circles starting every 30 ft and ending 0.4 ft later.
+light.example.text.aura.sideFields  = 4 cones with an arc of 90 degrees and range of 12.5, each one offset to cover a different region.
+
```
Comment (irisiflimsi) [inline src/main/resources/net/rptools/maptool/language/i18n.properties:871]: Editorial only: Why have justified "equals" here, when this isn't done anywhere else?

**Review by irisiflimsi — APPROVED** [review]
(I don't know, whether I should "approve" or simply comment. I'm going to do it anyway.)

**Comment by bubblobill** [comment]
@irisiflimsi All comments welcome. I've reworked this so many times in the last week I've lost some of my perspective of the whole. Since the main driver is to make the translation keys and strings easier to understand for the translation community, anything that makes it easier to read, understand, and maintain is worth noting.

File: src/main/resources/net/rptools/maptool/language/i18n.properties (hunk 5 above)
Comment (bubblobill) [reply src/main/resources/net/rptools/maptool/language/i18n.properties:871]: Residual effect of multiple edits and probably Notepad++ disagreeing with Spotless.

File: src/main/java/net/rptools/maptool/client/ui/campaignproperties/CampaignPropertiesDialog.java (hunk 4 above)
Comment (bubblobill) [reply src/main/java/net/rptools/maptool/client/ui/campaignproperties/CampaignPropertiesDialog.java:633]: Bwahahaha... you should have seen the long version.
Depending on how fast we get the new form-based UI for light/sight this version may not last very long.

File: src/main/java/net/rptools/maptool/client/ui/campaignproperties/CampaignPropertiesDialog.java (hunk 1 above)
Comment (bubblobill) [reply src/main/java/net/rptools/maptool/client/ui/campaignproperties/CampaignPropertiesDialog.java:590]: I can't format it as a link (preferred) because it won't actually open in a browser (or at least it didn't when I tried). I wanted it to stand out so people know what to copy-paste. I can't hard-code the colour because that may not be visible in some themes and linking to theme.css didn't work for utilising theme-sensitive styling like accent colour.
That left me with bold, italic, and underline as means of differentiation. Underline implies clickable which is inappropriate and bold fails my aesthetic text for the second part of a sentence. That left me with italic, the best of an admittedly stagnant pool of options.

File: src/main/java/net/rptools/maptool/client/ui/campaignproperties/CampaignPropertiesDialog.java (hunk 3 above)
Comment (bubblobill) [reply src/main/java/net/rptools/maptool/client/ui/campaignproperties/CampaignPropertiesDialog.java:817]: Easy done. Shall implement.

**Comment by bubblobill** [comment]
#2120

File: src/main/java/net/rptools/maptool/client/ui/campaignproperties/CampaignPropertiesDialog.java (hunk 1 above)
Comment (irisiflimsi) [reply src/main/java/net/rptools/maptool/client/ui/campaignproperties/CampaignPropertiesDialog.java:590]: Does it have to be formatted at all? This is the only place I saw, where formatting is part of the "value". The other sections have the formatting in the "container". E.g., here it could be "...<i>{sightLight.wikiLinkReferral}</i>..."

Comment (bubblobill) [reply src/main/java/net/rptools/maptool/client/ui/campaignproperties/CampaignPropertiesDialog.java:590]: In order to help keep the translation text context I'm passing it as a parameter.
So the translator sees something like "referral.wikilink = find more in the wiki online at {0}."
Because it is being injected into the sentence it needs to carry its own formatting if I want to keep the tags out of the translatable text.
Formatting is not necessary, but I want it, so I coded it to be kept out of the way of everyone except developers.

Comment (irisiflimsi) [reply src/main/java/net/rptools/maptool/client/ui/campaignproperties/CampaignPropertiesDialog.java:590]: I see, that part is never to be translated anyway. The chances are less than slim, that this would ever be referenced as wiki.rptools.info/index.php/es/introduccion_a_luces_y_vistas...

**Review by cwisniew — APPROVED** [review]
